apps/assistants/models.py

```diff
@@ -76,6 +87,33 @@ def supports_file_search(self):
     def tools_enabled(self):
         return len(self.tools) > 0
 
+    @transaction.atomic()
+    def create_new_version(self, *args, **kwargs):
+        from .sync import push_assistant_to_openai
+
+        version_number = self.version_number
+        self.version_number = version_number + 1
+        self.save(update_fields=["version_number"])
+        assistant_version = super().create_new_version(save=False, *args, **kwargs)
+        assistant_version.version_number = version_number
+        assistant_version.save()
+
+        assistant_version.files.set(self.files.all())
+        assistant_version.tools = self.tools.copy()
+        for tool_resource in self.tool_resources.all():
+            files = tool_resource.files.all()
+            if files:
+                new_tool_resource = ToolResources.objects.create(
+                    assistant=assistant_version,
+                    tool_type=tool_resource.tool_type,
+                )
+                new_tool_resource.files.set(files)
+                new_tool_resource.extra = tool_resource.extra
+                new_tool_resource.save()
+
+        push_assistant_to_openai(assistant_version, internal_tools=get_assistant_tools(assistant_version))
```

Comment:
The `assistant_id` doesn't seem to be cleared, in which case the versions will point to the working version at OpenAI. We'll need to clear the `assistant_id` just before calling `push_assistant_to_openai`. Reason being that the `assistant_id` field cannot be empty, so you wouldn't be able to clear it and save the assistant.